libnavui-androidauto/src/main/java/com/mapbox/androidauto/notification/CarNotificationInterceptor.kt

```diff
@@ -0,0 +1,135 @@
+package com.mapbox.androidauto.notification
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.car.app.model.CarColor
+import androidx.car.app.notification.CarAppExtender
+import androidx.car.app.notification.CarPendingIntent
+import androidx.core.app.NotificationCompat
+import com.mapbox.androidauto.R
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.TimeFormat
+import com.mapbox.navigation.base.formatter.DistanceFormatter
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.formatter.MapboxDistanceFormatter
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+import com.mapbox.navigation.core.trip.session.NavigationSessionStateObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+
+/**
+ * Register this observer using [MapboxNavigationApp.registerObserver]. As long as it is
+ * registered, the trip notification will be shown in the car.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class CarNotificationInterceptor internal constructor(
+    private val context: Context,
+    private val pendingOpenIntent: PendingIntent?,
+    private val setupNotification: NotificationCompat.Builder.() -> Unit,
+    private val setupCarAppExtender: CarAppExtender.Builder.() -> Unit,
+    private val idleExtenderUpdater: IdleExtenderUpdater,
+    private val freeDriveExtenderUpdater: FreeDriveExtenderUpdater,
+    private val activeGuidanceExtenderUpdater: ActiveGuidanceExtenderUpdater,
+) : MapboxNavigationObserver {
+
+    private var navigationSessionState: NavigationSessionState = NavigationSessionState.Idle
+    private var routeProgress: RouteProgress? = null
+
+    private val navigationSessionStateObserver = NavigationSessionStateObserver { sessionState ->
+        navigationSessionState = sessionState
+    }
+
+    private val routeProgressObserver = RouteProgressObserver { routeProgress ->
+        this.routeProgress = routeProgress
+    }
+
+    @JvmOverloads constructor(
+        context: Context,
+        contentIntent: Intent? = null,
```

Comment:
This is a somewhat generic intent, but it is also specific considering it is transformed into a CarPendingIntent.

I think we should go one way or another here. Surface a `PendingIntent` and make it clear that it is sent to the `CarAppExtender`; or make it specifically handle one use case.

I think we should make it specifically handle one use case. Optimizing for a simple usage.
``` kotlin
        CarNotificationInterceptor(
            carContext,
            MainCarAppService::class.java
        )
```

``` kotlin
    @JvmOverloads constructor(
        context: Context,
        openOnTapClass: Class<out CarAppService>,
    ) : this(
        context,
        CarPendingIntent.getCarApp(
            context,
            0,
            Intent(Intent.ACTION_VIEW).setComponent(ComponentName(context, openOnTapClass)),
            PendingIntent.FLAG_UPDATE_CURRENT
        ),
``